src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -873,24 +812,27 @@ private static void configurePublishConstraints(
             List<DependencyConstraint> publishConstraints) {
         ListProperty<DependencyConstraint> constraintsProperty =
                 GradleWorkarounds.fixListProperty(project.getObjects().listProperty(DependencyConstraint.class));
-        constraintsProperty.addAll(project.provider(Suppliers.memoize(() -> {
-            log.debug("Computing publish constraints for {} by resolving {}",
-                    configuration.get(),
-                    configurationForFiltering.get());
-            Set<ModuleIdentifier> modulesToInclude = configurationForFiltering
-                    .get()
-                    .getIncoming()
-                    .getResolutionResult()
-                    .getAllComponents()
-                    .stream()
-                    .map(ResolvedComponentResult::getModuleVersion)
-                    .filter(Objects::nonNull)
-                    .map(ModuleVersionIdentifier::getModule)
-                    .collect(Collectors.toSet());
-            return Collections2.filter(
-                    publishConstraints,
-                    constraint -> modulesToInclude.contains(constraint.getModule()));
-        })::get));
+        constraintsProperty.addAll(project.provider(
+                Suppliers.memoize(() -> {
+                            log.debug(
+                                    "Computing publish constraints for {} by resolving {}",
+                                    configuration.get(),
+                                    configurationForFiltering.get());
+                            Set<ModuleIdentifier> modulesToInclude =
+                                    configurationForFiltering
+                                            .get()
+                                            .getIncoming()
+                                            .getResolutionResult()
+                                            .getAllComponents()
+                                            .stream()
+                                            .map(ResolvedComponentResult::getModuleVersion)
+                                            .filter(Objects::nonNull)
+                                            .map(ModuleVersionIdentifier::getModule)
+                                            .collect(Collectors.toSet());
+                            return Collections2.filter(publishConstraints, constraint -> modulesToInclude.contains(
+                                    constraint.getModule()));
+                        })
+                        ::get));
```

Comment:
this whole thing is indented because of the trailing `::get`, however I think we can special-case this kind of thing and allow it to flow right after the method e.g. 
```java
Suppliers.memoize(() -> {
    ...
})::get
```